fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordValidationTest.java

```diff
@@ -0,0 +1,238 @@
+        // Delete a split
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            // If operating on the short record, #0 is the only split
+            // If operating on the long record, splits can be 1,2,3
+            final Tuple pkAndSplit = result.get(recordNumber).getPrimaryKey().add(splitNumber);
+            byte[] split = store.recordsSubspace().pack(pkAndSplit);
+            store.ensureContextActive().clear(split);
+            commit(context);
+        }
+
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            final FDBRecordStoreBase.RecordValidationOptions expected;
+            // When format version is 3 and the record is a short record, deleting the only split will make the record disappear
+            if ((splitNumber == 0) && (formatVersion == 3)) {
+                expected = FDBRecordStoreBase.RecordValidationOptions.RECORD_EXISTS;
+            } else {
+                expected = FDBRecordStoreBase.RecordValidationOptions.VALID_VALUE;
+            }
+            validateRecord(store, result.get(recordNumber), expected);
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsDeserialize [formatVersion = {0}]")
+    @MethodSource("formatVersions")
+    public void testValidateRecordsDeserialize(int formatVersion) throws Exception {
+        boolean splitLongRecords = true;
+        // Remove the last split from the record (#3) so that it fails to deserialize
+        int recordNumber = 1;
+
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> result = saveRecords(splitLongRecords, formatVersion, hook);
+        // Delete split #3
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            final Tuple pkAndSplit = result.get(recordNumber).getPrimaryKey().add(3);
+            byte[] split = store.recordsSubspace().pack(pkAndSplit);
+            store.ensureContextActive().clear(split);
+            commit(context);
+        }
+
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            validateRecord(store, result.get(recordNumber), FDBRecordStoreBase.RecordValidationOptions.DESERIALIZABLE);
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsMissingVersion [splitLongRecords = {0}]")
+    @BooleanSource
+    public void testValidateRecordsMissingVersion(boolean splitLongRecords) throws Exception {
```

Comment:
How is this supposed to behave with `SAVE_UNSPLIT_WITH_SUFFIX_FORMAT_VERSION` ?

Reply:
I didn't think this was a relevant test case here since the version is not saved with the record

Reply:
For version 5, we don't have version splits (that's version 6), so there wouldn't be a version split to delete, right?